File: algorithms/Deque/Easy/QueuesCRUDLinkedList/VSDQueuesCRUDLinkedList.py

class Node:
    def __init__(self,data):
        self.data=data 
        self.next=None 

class QueueLL:
    def __init__(self):
        self.front=None
        self.rear=None 

    def isempty(self):
        if not self.front and not self.rear:
            return True 
        return False

    def enqueue(self,data):
        if self.rear is None:
            self.rear=Node(data)
            self.front=self.rear 
        else:
            self.rear.next=Node(data)
            self.rear=self.rear.next 

    def dequeue(self):
        if self.front is None:
            return 
        else:
            temp=self.front.data 
            self.front=self.front.next 
            return temp
    
    def traverse(self):
        head=self.front 
        while head:
            print(head.data,end=" ")
            head=head.next 
        print()

queue=QueueLL()
queue.enqueue(10)
queue.enqueue(20)
queue.enqueue(30)
queue.enqueue(40)
queue.enqueue(50)
queue.enqueue(60)
queue.enqueue(70)

queue.traverse()
queue.dequeue()
queue.dequeue()
queue.dequeue()
queue.traverse()

File: algorithms/Deque/Easy/QueuesCRUDLinkedList/VSDQueuesCRUDLinkedList.java
// Copyrights to venkys.io
// For more programs visit venkys.io 
// Java program for Queue

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
    }
}

class Queue {
    Node head;
    Node tail;

    void enqueue(int data) {
        if (tail == null) {
            Node temp = new Node(data);
            tail = temp;
            head = temp;
        } else {
            Node temp = new Node(data);
            tail.next = temp;
            tail = temp;

        }
    }

    void dequeue() {
        if(head!=null){
            if(head==tail)tail=tail.next;
            head=head.next;        
        }
    }

    void print() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
        System.out.println();
    }
}

public class Main {
    public static void main(String[] args) {
        Queue q = new Queue();
        q.enqueue(10);
        q.enqueue(20);
        q.print();
        q.dequeue();
        q.dequeue();
        q.print();
        q.enqueue(30);
        q.enqueue(40);
        q.enqueue(50);
        q.print();
    }
}